posit-dev/positron
# ci: add ability to run additional test tags against PRs

### Intent
This PR enhances our test-based workflows by making them more modular, composable, and dispatchable (is that a word?). In addition, we can now specify additional tags with an ampersand (`&critical`, etc) anywhere in PR text to run additional tests against a pull request.

### Changes

Modular Workflows
* Split core tests (e2e-linux, e2e-windows, unit-tests, integration-tests) into standalone workflows.
   * Parent workflows can now combine these tests as needed.
   * Added parameters/inputs to e2e workflows for tagging and filtering e2e tests.
* Added bash script to handle combining Playwright tags to run a subset of tests:
   * Enables AND logic for tags where needed.
   * Handles “base tags” like `@win` or `@web`

Dynamic Tag Parsing
* Allows tags (`&critical`, etc) in PR descriptions to be parsed and included dynamically.
* Ensures at minimum that critical tests always run (even if not tagged)
* Added a bash script to handle parsing PR text to pass along to e2e test runner

### QA Notes
* Ran the full test suite and confirmed it passed and reports were available as expected.
* Once merged, I plan to manually verify each of the workflow dispatches works as expected, but I did try to test this as much as I could already.

### Screenshots

test-full-suite.yml dispatch
<img width="360" alt="full-suite" src="https://github.com/user-attachments/assets/7bbc55d2-c0cb-4976-b3f5-afb45bd65c7c">

test-e2e-linux.yml dispatch
<img width="354" alt="linux" src="https://github.com/user-attachments/assets/dff56e8b-9d9b-4cbb-bdd0-85fc059f7199">

test-e2e-windows.yml dispatch
<img width="351" alt="windows" src="https://github.com/user-attachments/assets/c760f80b-3e6a-4d86-b3c4-282b400cd5f1">

test-e2e-release.yml dispatch
<img width="359" alt="latest-release" src="https://github.com/user-attachments/assets/1b2f65e1-5fc6-4e7f-9626-9adf1b998c5e">
